### PR TITLE
Add support for pre-POSIX `tar` 

### LINKED ISSUE #76: Pre-POSIX.1-1988 (i.e. v7) tar header
Hi, is it possible to add the headers for Pre-POSIX.1-1988 (i.e. v7) tar recognition?

Reference here: https://en.wikipedia.org/wiki/Tar_(computing)#cite_note-2

Thanks in advice!

### PR DESCRIPTION
**Pull request by stroncium**
Fixes #76 

Replaced ustar magic check with checksum magic check and checksum validation.
Attempts to add some more magic failed, tar implementations don't seem to respect `pad wtih NUL` policy even for legacy mode.
Recommendations from https://en.wikipedia.org/wiki/Tar_(computing)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -16,6 +16,31 @@ function readUInt64LE(buf, offset = 0) {
...
+		return false;
+	}
+
+	let sum = 256;
+	let signedBitCount = 0;
+
+	for (let i = 0; i < 148; i++) {
+		const byte = buf[i];
+		sum += byte;
+		signedBitCount += byte & 0x80;
```
Comment (sindresorhus) [inline index.js]: Can you move `0x80` to a variable with a descriptive name?

File: index.js [hunk 2]
```diff
@@ -16,6 +16,31 @@ function readUInt64LE(buf, offset = 0) {
...
+	for (let i = 156; i < 512; i++) {
+		const byte = buf[i];
+		sum += byte;
+		signedBitCount += byte & 0x80;
+	}
+
+	const readSum = parseInt(buf.toString('utf8', 148, 154), 8);
+
+	return sum === readSum || ((sum - (signedBitCount << 1)) === readSum);
+};
```
Comment (sindresorhus) [inline index.js]: Can you add some comments in this function to explain what's going on?

File: index.js (hunk 1 above)
Comment (stroncium) [reply index.js]: But it's not a variable, it's bitmask.

Comment (stroncium) [reply index.js]: done

File: index.js (hunk 2 above)
Comment (stroncium) [reply index.js]: done

**Comment by sindresorhus** [comment]
Thanks. Looks good now :)

File: index.js [hunk 3]
```diff
@@ -16,6 +16,37 @@ function readUInt64LE(buf, offset = 0) {
...
+		sum += byte; // Add to sum
+		signedBitSum += byte & MASK_8TH_BIT; // Add signed bit to signed bit sum
+	}
+
+	// Skip checksum field
+
+	for (let i = 156; i < 512; i++) {
+		const byte = buf[i];
+		sum += byte; // Add to sum
+		signedBitSum += byte & MASK_8TH_BIT; // Add signed bit to signed bit sum
```
Comment (noisyui) [inline index.js]: Clear all lower bits except for the signed bit?

**Pull request merged**